llvm/utils/TableGen/RegisterInfoEmitter.cpp

```diff
@@ -977,7 +977,8 @@ void RegisterInfoEmitter::runMCDesc(raw_ostream &OS, CodeGenTarget &Target,
        << DiffSeqs.get(SubRegLists[i]) << ", " << DiffSeqs.get(SuperRegLists[i])
        << ", " << SubRegIdxSeqs.get(SubRegIdxLists[i]) << ", "
        << (Offset << RegUnitBits | FirstRU) << ", "
-       << LaneMaskSeqs.get(RegUnitLaneMasks[i]) << " },\n";
+       << LaneMaskSeqs.get(RegUnitLaneMasks[i]) << ", " << Reg.Constant
```

Comment:
Could you point out where this change manifests in the test case above?

Reply:
This change manifests by setting the `bool Constant` variable in MCRegisterDesc as true for constant registers. It can be observed when .inc files are generated. We go from elements in the array of MCRegisterDesc <Target>RegDesc[] looking like this:

`  {uint32_t Name, uint32_t SubRegs, uint32_t SuperRegs, uint32_t SubRegIndices, uint32_t RegUnits, uint16_t RegUnitLaneMasks}`
To this:
` {uint32_t Name, uint32_t SubRegs, uint32_t SuperRegs, uint32_t SubRegIndices, uint32_t RegUnits, uint16_t RegUnitLaneMasks, bool Constant}`
We then check if a given register is constant and skip it. I've precommited the V1-zero-dependency.s test to showcase the effect of this patch. Comparing the test in the two commits there is a higher IPC and the timeline view shows the zero register no longer creates dependencies when this patch is applied.

Reply:
Thanks, I missed that this was in TableGen and not part of a MCA view.